code/modules/gear_presets/synths.dm

```diff
@@ -12,10 +11,22 @@
 	access = get_access(ACCESS_LIST_GLOBAL)
 
 /datum/equipment_preset/synth/load_race(mob/living/carbon/human/new_human)
+	var/generation_selection = SYNTH_GEN_THREE
 	if(new_human.client?.prefs?.synthetic_type)
-		new_human.set_species(new_human.client.prefs.synthetic_type)
-		return
-	new_human.set_species(SYNTH_GEN_THREE)
+		generation_selection = new_human.client.prefs.synthetic_type
+	switch(generation_selection)
+		if(SYNTH_GEN_THREE)
+			new_human.set_species(SYNTH_GEN_THREE)
+			skills = /datum/skills/synthetic
+		if(SYNTH_GEN_TWO)
+			new_human.set_species(SYNTH_COLONY_GEN_TWO)
+			skills = /datum/skills/colonial_synthetic
+		if(SYNTH_GEN_ONE)
+			new_human.set_species(SYNTH_COLONY_GEN_ONE)
+			skills = /datum/skills/colonial_synthetic
+		else
+			new_human.set_species(SYNTH_GEN_THREE)
+			skills = /datum/skills/synthetic
```

Comment:
this is a singleton, we shouldn't really be altering the variables of it

why not just override /load_skills like

```dm
/datum/equipment_preset/synth/load_skills(mob/living/carbon/human/new_human, client/mob_client)
	new_human.allow_gun_usage = FALSE

	if(iscolonysynthetic(new_human) && !isworkingjoe(new_human))
		new_human.set_skills(/datum/skills/colonial_synthetic)
		return

	if(!mob_client)
		new_human.set_skills(/datum/skills/synthetic)
		return

	switch(mob_client.prefs.synthetic_type)
		if(SYNTH_GEN_ONE, SYNTH_GEN_TWO)
			new_human.set_skills(/datum/skills/colonial_synthetic)
		else
			new_human.set_skills(/datum/skills/synthetic)
```

you also won't need to double up code for /datum/equipment_preset/synth/survivor

Reply:
Awesome, thanks!